symfit/core/support.py

```diff
@@ -135,12 +137,49 @@ def parameters(names, **kwargs):
     Convenience function for the creation of multiple parameters. For more
     control, consider using ``symbols(names, cls=Parameter, **kwargs)`` directly.
 
+    The `Parameter` attributes `value`, `min`, `max` and `fixed` can also be provided
+    directly. If given as a single value, the same value will be set for all
+    `Parameter`'s. When a sequence, it must be of the same length as the number of
+    parameters created.
+
+    Example::
+        x1, x2 = parameters('x1, x2', value=[2.0, 1.3], min=0.0)
+
     :param names: string of parameter names.
         Example: a, b = parameters('a, b')
-    :param kwargs: kwargs to be passed onto :func:`sympy.core.symbol.symbols`
+    :param kwargs: kwargs to be passed onto :func:`sympy.core.symbol.symbols`.
+        `value`, `min` and `max` will be handled separately if they are sequences.
     :return: iterable of :class:`symfit.core.argument.Parameter` objects
     """
-    return symbols(names, cls=Parameter, seq=True, **kwargs)
+    sequence_fields = ['value', 'min', 'max', 'fixed']
+    sequences = {}
+    for attr in sequence_fields:
+        # if attr in kwargs:
+        try:
+            len(kwargs[attr])
+        except (TypeError, KeyError):
+            # Not iterable
+            pass
+        else:
+            sequences[attr] = kwargs.pop(attr)
+
+    if 'min' in sequences and 'max' in sequences:
+        for min, max in zip(sequences['min'], sequences['max']):
+            if min > max:
+                raise ValueError('The value of `min` should be less than or'
+                                 ' equal to the value of `max`.')
+
+    ans = symbols(names, cls=Parameter, seq=True, **kwargs)
+    for key, values in sequences.items():
+        if not isinstance(values, repeat) and len(values) != len(ans):
+            raise ValueError(
+                '`len` of keyword-argument `{}` does not match the number of '
+                '`Parameter`s created.'.format(attr)
+            )
```

Comment:
I wouldn't test for this case I think. `values` might be iterators/generators/whatevers that don't have a `len`.
Instead, consider the following alternative:
```python
params = ans
extra_attrs = []
attrs, vals = zip(*sequences.items())
attr_per_param = [{k: val for k, val in zip(attrs, vs)} for vs in zip(*vals)]
if len(attr_per_param) != len(params):
    raise
for param, attrs in zip(params, attr_per_param):
    for name, val in attrs.items():
        setattr(param, name, val)
```
Look up how zip deals with too short sequences though. Now you could maybe end up with some params getting an e.g. `max` but not all.

Reply:
Zip stops as soon as one of them terminates, so some explicit checking is required. However, I think I have a nice solution, soon to be pushed

Reply:
As it is now it's better, but still not there yet I think:
```python
sequences = {'min': iter([1, 2]), 'max': [1, 2, 3]}
names = ['a', 'b', 'c']
```

Also, make an if/else instead of assert. 

```python
params = ans
extra_attrs = []
attrs, vals = zip(*sequences.items())
attr_per_param = [{k: val for k, val in zip(attrs, vs)} for vs in zip(*vals)]
if len(attr_per_param) != len(params):
    raise
for param, attrs in zip(params, attr_per_param):
    if attrs.keys() != sequences.keys():
        raise
    for name, val in attrs.items():
        setattr(param, name, val)
```